tableauserverclient/namespace.py

```diff
@@ -1 +1,30 @@
-NAMESPACE = {'t': 'http://tableau.com/api'}
+from xml.etree import ElementTree as ET
+import re
+
+OLD_NAMESPACE = 'http://tableausoftware.com/api'
+NEW_NAMESPACE = 'http://tableau.com/api'
+NAMESPACE_RE = re.compile(r'\{(.*?)\}')
+
+
+class Namespace(object):
+    def __init__(self):
+        self._namespace = {'t': NEW_NAMESPACE}
+        self._detected = False
+
+    def __call__(self):
+        return self._namespace
+
+    def detect(self, xml):
+        if self._detected:
+            return
+
+        if not xml.startswith(b'<?xml'):
+            return  # Not an xml file, don't detect anything
+
+        root = ET.fromstring(xml)
+        matches = NAMESPACE_RE.match(root.tag)
+        if matches:
+            detected_ns = matches.group(1)
+            if detected_ns in (OLD_NAMESPACE, NEW_NAMESPACE):
```

Comment:
Is there a reason we don't throw here if we don't know what it is? Won't we fail in an odd way later?

Reply:
Not every request necessarily returns xml (deletes for example). 

This just skips it an checks on the next request. Once it succeeds we cache the result and don't check again.

We default to the new namespace so nothing should fail

Reply:
no real reason.  I'll add a new exception